packages/fuels-signers/src/wallet.rs

```diff
@@ -73,21 +73,14 @@ pub enum WalletError {
     /// Error propagated from the hex crate.
     #[error(transparent)]
     Hex(#[from] hex::FromHexError),
-    /// Error propagated by IO operations
-    #[error(transparent)]
-    Io(#[from] std::io::Error),
     /// Error propagated by parsing of a slice
     #[error("Failed to parse slice")]
     Parsing(#[from] std::array::TryFromSliceError),
     #[error("No provider was setup: make sure to set_provider in your wallet!")]
     NoProvider,
-    #[error("Provider error: {0}")]
-    ProviderError(#[from] ProviderError),
     /// Keystore error
     #[error(transparent)]
     KeystoreError(#[from] KeystoreError),
-    #[error("invalid mnemonic word count (expected 12, 15, 18, 21, 24, found `{0}`")]
-    InvalidMnemonicWordCount(usize),
```

Comment:
Why did you remove it? Even if it is not used currently, it would probably be in the future? Or is it to "merge it" with `MnemonicError`?

Reply:
I assumed that it was used at some point and now lost its purpose, the YAGNI way would be to remove it, but I don't mind keeping it. Its your call.

Reply:
Yeah I guess merging it into `MnemonicError` is fine either way, trash it!